pkg/skaffold/build/docker/errors.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/pkg/jsonmessage"
+
+	sErrors "github.com/GoogleContainerTools/skaffold/pkg/skaffold/errors"
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+func newBuildError(err error) error {
```

Comment:
This function needs some documentation!
```suggestion
// newBuildError turns Docker-specific errors into actionable errors.  The input errors are assumed to be
// from the Skaffold docker invocation.
func newBuildError(err error) error {
```

Reply:
added it now.